ITE soc/riscv-ite/it8xxx2: move config to soc from board

RuibinChang opened the pull request:
Make config conditional and move to soc Kconfig.defconfig.series
from it8xxx2_evb_defconfig.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

RuibinChang committed:
ITE soc/riscv-ite/it8xxx2: move config to soc from board

Make config conditional and move to soc Kconfig.defconfig.series
from it8xxx2_evb_defconfig.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>

RuibinChang commented:
As @MaureenHelm suggestion in tachometer PR (https://github.com/zephyrproject-rtos/zephyr/pull/39993).

NOTE: After tachometer PR is merged, this PR should be rebased and solved conflict then could be merged .

MaureenHelm commented on soc/riscv/riscv-ite/it8xxx2/Kconfig.defconfig.series:
```diff
@@ -16,19 +16,57 @@ config ITE_IT8XXX2_INTC
+	depends on ADC
+
+config SOC_FLASH_ITE_IT8XXX2
+	default y
```
`depends on FLASH`

RuibinChang replied:
Done

RuibinChang replied:
So if I want to build the it8xxx2_evb with tachometer driver, then I should add the CONFIG_SENSOR=y in samples/boards/it8xxx2_evb/prj.conf (if the path exists) , right?

MaureenHelm replied:
Yes

MaureenHelm reviewed (changes requested):
Thanks for following up on this, it looks good. The next step is to add a second commit that removes the following from `boards/riscv/it8xxx2_evb/it8xxx2_evb_defconfig`:
- CONFIG_WATCHDOG=y
- CONFIG_KSCAN=y
- CONFIG_I2C=y
- CONFIG_PWM=y

The net result after the two commits will be that these drivers are no longer enabled in hello_world and the right SoC-specific drivers are enabled in applications that need them.

keith-zephyr reviewed (approved):
LGTM with Maureen's requested change.

brockus-zephyr reviewed (approved):
LGTM with Maureen's request.

RuibinChang committed:
ITE boards/it8xxx2_evb: remove CONFIG_* not for every application

These drivers are no longer enabled for every application,
even hello_world. The right SoC-specific drivers are
enabled in applications that need them.

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>
